mmi-cookbooks/metrics-filebeat
# Add unit and integration tests, set up 2 new prospectors

# What
This add unit and integration tests to this wrapper cookbook.  Unit tests are just testing resources, the integration tests do a package install, service running and some light content checking of the prospector files - in addition a couple of them have negative tests.

This also adds prospectors for repose application logs (in the ingest and query node configs) and adds an elasticsearch recipe which will ship curator logs to be indexed and stored.

Bump version to 0.2.0.  Set up travisCI to run multiple test suites using Docker.

# How
Added unit and integration tests under the test directory.  Added a spec matcher file to get unit tests for filebeat_prospector to work.  Updated kitchen test into several suites so that integration tests can be run on individual types.  

## How to test
Unit tests can be run with $ chef exec rspec test/unit
Integration tests could be run with $ rake integration  (this would take a while to run all 6 suites)
or individually using $ kitchen verify <type>

# Why
Increasing test coverage (and skills) is good.
Wanted to add a couple of more prospectors.
Hopefully some of this can be reused for MaaS setting up filebeat in the near term.
